_posts/2023-10-02-intellij-quarkus-recap.adoc

```diff
@@ -0,0 +1,178 @@
+image:java-validation-graphql-directives.png[GraphQL directives placement validation]
+
+- Quarkus `BuildItem` subclasses are checked whether they're `final` or `abstract`:
+image:java-builditem-validation.png[BuildItem validation]
+
+=== Severity mappings
+We've tried to bind some Microprofile and Qute diagnostics reported by the language servers to IntelliJ's Inspections settings, as it felt more natural than having specialized preference pages. But, since this is a bit of a hack, the inspections settings are mostly ignored, except for a couple severity settings. The limitations are described in each inspection description:
+
+image:inspections-severity-mapping.png[Inspection severity mapping]
+
+In most cases:
+
+- Scope: values are ignored
+- Severity: only Error and (Weak) Warning are respected. Other values mean no errors will be reported
+- Highlighting in Editor: values are ignored
+
+=== Quick-fixes updating settings
+
+We've introduced quick fixes that allow you to disable validation for certain errors. Useful, for instance, when you find false-positive errors add too much noise to your development workflow.
+
+Unassigned `@ConfigProperty` properties (i.e not declared in `application.properties`) in java files, or unknown properties (seemingly unused) declared in `application.properties` can now be excluded from validation via a quick-fix, e.g:
+
+image:exclusions-quickfix.png[Quick-fix to exclude property from validation]
+
+You can then update those exclusions in `Preferences | Editor | Inspections | MicroProfile | Java files | Unassigned properties` or `Preferences | Editor | Inspections | MicroProfile | Properties files | Unassigned properties`
+
+image:exclusions-settings.png[Inspections settings]
+
+
+Similarly you can exclude a particular Qute template from validation, or its entire module
+
+image:exclude-qute-validation.png[Exclude Qute template from validation]
+
+You can then head over to `Preferences | Editor | Inspections | Qute | Templates | Validation`, 
+
+image:qute-validation-settings.png[Qute validation settings]
+
+
+== Miscellaneous improvements
+- The Quarkus project wizard, on top of being super snappy, now offers easy (de)selection of extensions by double-clicking on them.
+- Projects in WSL2 are now supported.
+- Run configurations have been renamed to `Quarkus Dev Mode`, to minimize confusion with IntelliJ Ultimate's built-in Quarkus support.
+- Vert.x reactive routes, JAX-RS and Renarde endpoints are now shown as implicitly used, as well as Quarkus `@BuildStep`-annotated classes and `@Observer`-annotated methods.
+
+
+== Looking forward
+
+Quarkus' "developer joy" mantra obviously applies to your time spent in the IDE, that's what the Red Hat Developer team is vyying for. As we reflect on the past five months of releases, we're pretty excited about the progress we've made in the Quarkus development experience in IntelliJ IDEA. 
```

Comment:
- "and that's what the Red Hat Developer team"
- Apparently it's "vying"
- "progress we've made on the Quarkus development experience in IntelliJ IDEA"

Reply:
done